pkg/i2gw/providers/istio/converter.go

```diff
@@ -289,10 +291,36 @@ func (c *converter) convertGateway(gw *istioclientv1beta1.Gateway, fieldPath *fi
 	}, nil
 }
 
-func (c *converter) convertVsHTTPRoutes(virtualService metav1.ObjectMeta, istioHTTPRoutes []*istiov1beta1.HTTPRoute, allowedHostnames []string, fieldPath *field.Path) ([]*gatewayv1.HTTPRoute, field.ErrorList) {
+var hostnameRegexp = regexp.MustCompile(`^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`)
+
+// convertHostnames set istio hostnames as is, without extra filters.
+// If it's not a fqdn, it would be rejected by K8S API implementation
+func convertHostnames(hosts []string) []gatewayv1.Hostname {
+	var resHostnames []gatewayv1.Hostname
+	for _, host := range hosts {
+		// '*' is valid in istio, but not in HTTPRoute
+		if !hostnameRegexp.MatchString(host) {
+			klog.Infof("ignoring host %s, which is not allowed in Gateway API HTTPRoute", host)
```

Comment:
I'd give it severity warning
Same for line 309

Reply:
updated